impress-org/givewp
# Feature: add phone number block/field to the visual form builder and donation forms

Resolves [GIVE-532]

## Description



This PR adds a new Phone block to the visual form builder to allow admins to set up a form to collect the donor's phone numbers.  When the donation is finished, the phone number is displayed on the confirmation page under the "Donor Details" section, it also is displayed in the edit donation screen in the wp-admin area.

Also, this PR prevents a donor with a phone number saved previously from getting this value overwritten in cases where a new donation is made with a new number - instead of overwriting the previous value, we save the new phone number only in the donation meta. ~~instead of overwriting the previous value, we save the new phone number in the "additional phones" property.~~

## Affects



- The Donor model
- The Donation Form model
- The Visual Form Builder

## Visuals



![image](https://github.com/impress-org/givewp/assets/16308864/43ef9af1-1560-4113-a5b5-f923eefe4d1a)

![image](https://github.com/impress-org/givewp/assets/16308864/44b32f5b-24f6-40a8-b875-515cfd52ce18)

![image](https://github.com/impress-org/givewp/assets/16308864/4a2ce6ee-de38-489f-9225-8b549f0dcfb0)

![image](https://github.com/impress-org/givewp/assets/16308864/9cb833ed-7313-4657-9a8c-5dad160af5c6)


## Testing Instructions



1. Create a v3 form;
2. Add the new Phone block;
3. Make a test donation;
4. On the confirmation page, check if the phone is visible in the donor details;
5. On the donation edit page, check if the phone is displayed in the donor details section.

## Pre-review Checklist






[GIVE-532]: https://stellarwp.atlassian.net/browse/GIVE-532?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
